### ISSUE TITLE
Investigate the dual wielding CalcChanceToHitGun() code

### ISSUE THREAD

**Issue by Manuel-K** [ja2-stracciatella/ja2-stracciatella#700]
This came up in #495:

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/master/src/game/Tactical/Weapons.cc#L2169

I do not understand the that code block at all. The logic seems to be:
If we do not use a two-handed weapon, we assume that we are dual-wielding and:
- give a penalty to SMGs.
- if the soldier is not ambidextrous and `!IsValidSecondHandShot()` is true penalize the hit chance by `AIM_PENALTY_DUAL_PISTOLS`.

The second part seems correct, because `IsValidSecondHandShot()` checks if the soldier has two weapons. 
The first part does not do that. The `TODO0007` note mentions this as well: “Deduct AIM_PENALTY_SMG when firing a SMG even when second hand is free?”

Note: The questionable code is part of the original source code release.

**Comment by JAsmine-ja2** [comment]
I'm not sure if this is relevant, but AFAIR, when holding a one-handed gun in one hand and an item (e.g. a break light) in the other hand, the CtH is penalized to simulate that the merc is holding the gun with only one hand.

**Comment by DSeyka** [comment]
Not an expert on reading the code but it seems like SMGs are being given a penalty in all cases (so even when second hand is empty), which is probably not intended.

**Comment by lynxlynxlynx** [comment]
Not a weapon expert, but since SMGs have longer barrels than guns, I don't see a good reason why they would be less accurate (should be more at the same distance), if both are fired two-handed. I could see the penalty making sense for one-handed firing (including dual-wielding), since SMGs are heavier.

**Comment by JAsmine-ja2** [comment]
Agree. Dual wielding SMGs resulting in slightly lower CtH compared to dual wielding pistols seems okay to me.
Penalizing SMGs when firing it two-handed seems wrong to me.

**Comment by DSeyka** [comment]
Well, it hardly makes any sense to me to apply a flat penalty to weapon type in the code; it'd make more sense to lower item stats themselves if that was the intention.
I'm tempted to say that it's an incomplete feature which should probably work like JAsmine-ja2 describes.
